src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.cs

```diff
@@ -0,0 +1,467 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Numerics.Tensors
+{
+    /// <summary>Performs primitive tensor operations over spans of memory.</summary>
+    public static class TensorPrimitives
+    {
+        /// <summary>Computes the element-wise result of: <c><paramref name="x" /> + <paramref name="y" /></c>.</summary>
+        /// <param name="x">The first tensor, represented as a span.</param>
+        /// <param name="y">The second tensor, represented as a span.</param>
+        /// <param name="destination">The destination tensor, represented as a span.</param>
+        /// <exception cref="ArgumentException">Length of '<paramref name="x" />' must be same as length of '<paramref name="y" />'.</exception>
+        /// <exception cref="ArgumentException">Destination is too short.</exception>
+        /// <remarks>This method effectively does <c><paramref name="destination" />[i] = <paramref name="x" />[i] + <paramref name="y" />[i]</c>.</remarks>
+        public static void Add(ReadOnlySpan<float> x, ReadOnlySpan<float> y, Span<float> destination)
+        {
+            if (x.Length != y.Length)
+            {
+                ThrowHelper.ThrowArgument_SpansMustHaveSameLength(nameof(x), nameof(y));
+            }
+
+            if (x.Length > destination.Length)
+            {
+                ThrowHelper.ThrowArgument_DestinationTooShort();
+            }
+
+            for (int i = 0; i < x.Length; i++)
+            {
+                destination[i] = x[i] + y[i];
+            }
```

Comment:
I assume the plan is to get the functionality merged and to then follow-up with vectorization in time for 8.0 as well?

Let me know if you want to divvy that up at all or if you're going to tackle it all yourself.  Presumably we'll end up wanting a `Vector<T>` implementation for downlevel and a `Vector128/256/512<T>` for .NET 8+.

Reply:
Yes. The general plan is to get this first bit (covering element-wise operations) in.

Work on vectorizing them can then start while the vector-based tensor operations (Signmoid, Product, Sum, etc) get their naive implementations in.

@michaelgsharp and I will be splitting the work, but you're more than welcome to join in and help as well!

Reply:
Getting the naive implementation in gives us both a scalar fallback and a baseline against which correctness of the SIMD implementation can be validated.

I should have tests up here in just a little bit.

Reply:
> @michaelgsharp and I will be splitting the work, but you're more than welcome to join in and help as well!

Happy to help